src/mach/compact_unwind_info.rs

```diff
@@ -0,0 +1,1503 @@
+//! Support for the "compact unwinding format" used by Apple platforms,
```

Comment:
I love the documentation! So this file will take me a bit of time to review, thankfully it looks like half of it is documentation. It might also help to get another expert to review, any help appreciated :) I will probably hold off doing a review until this PR is ready to be non WIP though, so please do let me know when you think it's ready :)

Reply:
I have been iterating on this back in [the original PR on symbolic](https://github.com/getsentry/symbolic/pull/372/) and I believe it is in its ~final form, modulo what everyone wants me to do with [the outstanding TODOs](https://github.com/getsentry/symbolic/pull/372/#issuecomment-831357054) that can potentially be deferred for followups.

Just to avoid constantly synching things over and over, I'd like all review to go on there and then when everyone's happy I'll port the impl back over to goblin and we can discuss the implementation/signature of the entry point in `mach`.

The large outstanding question for me for integrating what I have into goblin is how to, using only goblin's API, detect the architecture (x86, x64, ARM64) which is otherwise not specified in the unwind_info section but determines how opcodes are interpreted.